### ISSUE TITLE
[Doc-feat] Add comparisons of point cloud filtrations

### ISSUE THREAD

**Issue by DavidLapous** [GUDHI/gudhi-devel#1220]
Something that we noticed with Vincent : 
It could be a good idea to add comparisons between methods, and in particularly point clouds filtrations, [à la sklearn](https://scikit-learn.org/stable/modules/clustering.html#id24).
The idea would be that in the point cloud filtration doc page, we add a few little examples, with the respective triangulations of Rips, Alpha, Witness, Edge collapsed rips, etc. at specific filtration values to intuitively show the differences. 
Adding pros and cons of each (with refs) is also a good idea (e.g., Delaunay is good for Euclidean low-dim spaces, sparse rips for large non-eudlidean point clouds, ...)
